x-pack/legacy/plugins/monitoring/public/components/alerts/configuration/configuration.tsx

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { ReactNode } from 'react';
+import { kfetch } from 'ui/kfetch';
+import { EuiSteps } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { ActionResult } from '../../../../../actions/server/types';
+import { ALERT_ACTION_TYPE_EMAIL } from '../../../../common/constants';
+import { getMissingFieldErrors } from '../../../lib/form_validation';
+import { Step1 } from './step1';
+import { Step2 } from './step2';
+import { Step3 } from './step3';
+
+export interface AlertsConfigurationProps {
+  emailAddress: string;
+  onDone: Function;
+}
+
+export interface StepResult {
+  title: string;
+  children: ReactNode;
+  status: any;
+}
+
+export interface AlertsConfigurationForm {
+  email: string | null;
+}
+
+export const NEW_ACTION_ID = '__new__';
+
+export const AlertsConfiguration: React.FC<AlertsConfigurationProps> = (
+  props: AlertsConfigurationProps
+) => {
+  const { onDone } = props;
+
+  const [emailActions, setEmailActions] = React.useState<ActionResult[]>([]);
+  const [selectedEmailActionId, setSelectedEmailActionId] = React.useState('');
+  const [editAction, setEditAction] = React.useState<ActionResult | null>(null);
+  const [emailAddress, setEmailAddress] = React.useState(props.emailAddress);
+  const [formErrors, setFormErrors] = React.useState<AlertsConfigurationForm>({ email: null });
+  const [showFormErrors, setShowFormErrors] = React.useState(false);
+  const [isSaving, setIsSaving] = React.useState(false);
+  const [saveError, setSaveError] = React.useState('');
+
+  React.useEffect(() => {
+    async function fetchData() {
+      await fetchEmailActions();
+    }
+
+    fetchData();
+  }, []);
+
+  React.useEffect(() => {
+    setFormErrors(getMissingFieldErrors({ email: emailAddress }, { email: '' }));
+  }, [emailAddress]);
+
+  async function fetchEmailActions() {
+    const kibanaActions = await kfetch({
```

Comment:
Any chance we can replace **all** instance of `kfetch` with `core.http`? it'll be easier for me to migrate if your PR goes in before my NP PR. Not really a big deal though, I can just create a `kfetch` shim if anything

Reply:
Do you have any examples of how I can access this in our legacy plugin right now?

Reply:
It should be as simple as:
```javascript
import { npStart } from 'ui/new_platform';
npStart.core.http.fetch(...)
```

Which we will then ref from the context instead of `ui/*`.

Up to you though, I can easily create a shim for it